### PR TITLE
Kyma function spec

### PR DESCRIPTION
**Pull request by MichalKalke**
**Description**

Changes proposed in this pull request:

- Update `FunctionSpec` properties in `function.go` file
- Update `synchronise` function in `synchronise.go` to meet with new `FunctionSpec` fields

**Related issue(s)**
[#1314](https://github.com/kyma-project/cli/issues/1314)

### CODE REVIEW

File: function/pkg/resources/unstructured/function.go [hunk 1]
```diff
@@ -36,21 +36,21 @@ func NewFunction(cfg workspace.Cfg) (unstructured.Unstructured, error) {
 }
 
 func newGitFunction(cfg workspace.Cfg) (out unstructured.Unstructured, err error) {
-	repository := cfg.Name
 	if cfg.Source.Repository != "" {
```
Comment (anoipm) [inline function/pkg/resources/unstructured/function.go:39]: Why we have `if` with empty body?

File: function/pkg/resources/unstructured/function_test.go [hunk 2]
```diff
@@ -735,10 +763,13 @@ func Test_newGitFunction(t *testing.T) {
 								workspace.ResourceNameMemory: "10M",
 							},
 						},
-						"source":    "test-name",
-						"baseDir":   "test-base-dir",
-						"reference": "test-reference",
-						"type":      "git",
+						"source": map[string]interface{}{
+							"gitRepository": map[string]interface{}{
+								"url":       "",
```
Comment (anoipm) [inline function/pkg/resources/unstructured/function_test.go:768]: Why "url" is empty? Other values are filled in this test.

File: function/pkg/resources/unstructured/function.go (hunk 1 above)
Comment (MichalKalke) [reply function/pkg/resources/unstructured/function.go:39]: Removed unused `if`

File: function/pkg/resources/unstructured/function_test.go (hunk 2 above)
Comment (MichalKalke) [reply function/pkg/resources/unstructured/function_test.go:768]: Fixed

File: function/pkg/workspace/workspace_test.go [hunk 3]
```diff
@@ -368,16 +373,16 @@ func inlineClient(ctrl *gomock.Controller, name, namespace string) client.Client
 	return result
 }
 
-func gitClient(ctrl *gomock.Controller, name, namespace string) client.Client {
+func gitClient(ctrl *gomock.Controller, namespace string) client.Client {
```
[the remark this answers is not in the document]
Comment (MichalKalke) [reply function/pkg/workspace/workspace_test.go]: Using `name` again instead of "test"

**Review by anoipm — APPROVED** [review]

**Pull request merged**